### PR TITLE
Fix #1980 Optimise hints_summary.xml

### LINKED ISSUE #1951: Optimise hints_summary.xml
This issue is tracking the optimisation work of `hints_summary.xml`

1. Make sure that the layout uses `ConstraintLayout` and there is minimal nesting of views. If this view already uses `ConstraintLayout` then it's fine.

2. Make sure that all textviews use the style from `styles.xml` file like `Heading1`, `TextFieldLabel`, `Body`, etc. whichever suits best.

While making the above changes the UI should not get affected and the end result should still be same as before (or an optimised version of UI).

If there are multiple files with the same name, update them all.

## References
Mobile Portrait: https://xd.adobe.com/view/e8aa4198-3940-47f9-514a-f41cc54457f6-9e9b/grid/
Mobile Landscape: https://xd.adobe.com/view/ee9e607b-dbd6-4372-48dc-b687d32af3da-98af/grid/
Tablet: https://xd.adobe.com/view/d405de00-a871-4f0f-73a0-f8acef30349b-a234/grid/
Wiki: https://github.com/oppia/oppia-android/wiki/Working-on-UI

### LINKED ISSUE #1980: Optimize code in administrator_controls_account_actions_view.xml
This issue is tracking the optimization work of `administrator_controls_account_actions_view.xml`

1. Make sure that the layout uses `ConstraintLayout` and there is minimal nesting of views. If this view already uses `ConstraintLayout` then it's fine.

2. Make sure that all TextViews use the style from `styles.xml` file like `Heading1`, `TextFieldLabel`, `Body`, etc. whichever suits best.

While making the above changes the UI should not get affected and the end result should still be the same as before (or an optimized version of UI).

If there are multiple files with the same name, update them all.

## References
Mobile Portrait: https://xd.adobe.com/view/e8aa4198-3940-47f9-514a-f41cc54457f6-9e9b/grid/
Mobile Landscape: https://xd.adobe.com/view/ee9e607b-dbd6-4372-48dc-b687d32af3da-98af/grid/
Tablet: https://xd.adobe.com/view/d405de00-a871-4f0f-73a0-f8acef30349b-a234/grid/
Wiki: https://github.com/oppia/oppia-android/wiki/Working-on-UI

### PR DESCRIPTION
**Pull request by secjedi**
Fixes #1951: Style the TextViews and Buttons in hint_summary.xml (landscape, portrait and tab mode) using styles defined in the styles.xml file.
The views in the layout already uses constraint layout 
Reviewer: @rt4914

## Checklist
- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

File: app/src/main/res/layout-land/hints_summary.xml [hunk 1]
```diff
@@ -10,106 +10,105 @@
...
 
-     <variable
+    <variable
       name="viewModel"
       type="org.oppia.android.app.hintsandsolution.HintsViewModel" />
   </data>
 
-    <LinearLayout
-      android:id="@+id/hint_summary_container"
+  <LinearLayout
```
Comment (Sarthak2601) [inline app/src/main/res/layout-land/hints_summary.xml]: We're aiming to flatten the view hierarchy by replacing the linear layout here by constraint layout. Try approaching that way.

File: app/src/main/res/layout-land/hints_summary.xml [hunk 2]
```diff
@@ -10,106 +10,105 @@
...
-          android:id="@+id/hints_and_solution_summary"
-          android:layout_width="wrap_content"
-          android:layout_height="wrap_content"
-          android:layout_gravity="start"
-          android:fontFamily="sans-serif"
-          android:textColor="@color/oppiaPrimaryText"
-          android:textSize="16sp" />
-      </LinearLayout>
+        android:layout_gravity="start"
+        android:fontFamily="sans-serif"
```
Comment (Sarthak2601) [inline app/src/main/res/layout-land/hints_summary.xml]: You can remove `android:fontFamily="sans-serif"`. It's already getting implemented using the style.

File: app/src/main/res/layout-land/hints_summary.xml [hunk 3]
```diff
@@ -10,106 +10,105 @@
...
-          android:layout_marginBottom="12dp"
-          android:fontFamily="sans-serif-medium"
-          android:textColor="@color/oppiaPrimaryText"
-          android:textSize="20sp"
-          app:layout_constraintStart_toStartOf="parent"
-          app:layout_constraintTop_toTopOf="parent" />
+      <Button
+        android:id="@+id/reveal_hint_button"
+        style="@style/StateButtonActive"
+        android:layout_width="wrap_content"
```
Comment (Sarthak2601) [inline app/src/main/res/layout-land/hints_summary.xml]: You can remove all the features that are getting enforced through the style. Things like `layout_width`, `layout_height` and `textColor` can be removed.

**Review by Sarthak2601 — CHANGES_REQUESTED** [review]
Good work @secjedi. I took a first pass at this and suggested a few things regarding the overall view hierarchy and style enforcement. Please make sure that the suggestions are incorporated in all the files.

File: app/src/main/res/layout-land/hints_summary.xml [hunk 4]
```diff
@@ -10,106 +10,105 @@
...
+      android:layout_marginEnd="72dp"
+      android:layout_marginBottom="20dp"
+      android:orientation="horizontal">
 
-      <androidx.constraintlayout.widget.ConstraintLayout
-        android:layout_width="match_parent"
+      <TextView
+        android:id="@+id/hint_title"
+        style="@style/Heading2"
+        android:layout_width="wrap_content"
```
Comment (Sarthak2601) [inline app/src/main/res/layout-land/hints_summary.xml]: Attributes like `layout_width`, `layout_height` and `textColor` can also be removed as they are getting enforced by the style. 
Remember, the style `Heading1` has a parent `TextView.Common1` which also enforces some attributes. Ditto elsewhere.

File: app/src/main/res/layout-land/hints_summary.xml (hunk 1 above)
Comment (secjedi) [reply app/src/main/res/layout-land/hints_summary.xml]: Thank you for the review @Sarthak2601. 
Do you mean every Linear layout should be replaced with constraint layout?

Comment (Sarthak2601) [reply app/src/main/res/layout-land/hints_summary.xml]: We're trying to get the most optimised UI here. So try to flatten the view hierarchy as much as you can without disrupting the existing design. That being said, if you think you can change both-give it a shot and if things go south, at least replace the outermost linear layout.

Comment (secjedi) [reply app/src/main/res/layout-land/hints_summary.xml]: Ok, I think I understand now. It is just fir this layer

Comment (secjedi) [reply app/src/main/res/layout-land/hints_summary.xml]: Done

File: app/src/main/res/layout-land/hints_summary.xml (hunk 2 above)
Comment (secjedi) [reply app/src/main/res/layout-land/hints_summary.xml]: Done

File: app/src/main/res/layout-land/hints_summary.xml (hunk 4 above)
Comment (secjedi) [reply app/src/main/res/layout-land/hints_summary.xml]: Done

**Comment by secjedi** [comment]
Done

**Comment by BenHenning** [comment]
@secjedi can you please update the title & PR description to correspond to #1980 instead of #1951?

**Comment by BenHenning** [comment]
Thanks @secjedi. Note that we format our PRs as: ``Fix #1234: Thing that changed`` (note the ``:`` that's missing in your title). Can you add that please? Also, shouldn't the title & PR description mention ``administrator_controls_account_actions_view.xml`` not ``hints_summary.xml``?

@rt4914 deferring this one to you, though feel free to send my way if you need to load balance.

**Comment by rt4914** [comment]
@secjedi Do you need to close this PR as this was recreated here: #2006

??

**Comment by rt4914** [comment]
@secjedi Please mention the reason why it was closed.

**Pull request closed without merging**